common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.util;
+
+import com.ibm.icu.text.StringSearch;
+
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * Static entry point for collation-aware expressions (StringExpressions, RegexpExpressions, and
+ * other expressions that require custom collation support), as well as private utility methods for
+ * collation-aware UTF8String operations needed to implement .
+ */
+public final class CollationSupport {
+
+  /**
+   * Collation-aware string expressions.
+   */
+
+  public static class Contains {
+    public static boolean contains(final UTF8String l, final UTF8String r, final int collationId) {
```

Comment:
Do we need to repeat class name in every method?
Can't you say something like:
Contains.exec(...)
Contains.genCode(...)
Contains.binaryExec(...)
Contains.lowercaseExec(...)
Contains.icuExec(...)

instead of
Contains.contains(...)

My hope is still that we will be able to add some base classes in future that will unify parts of this logic for some expressions. So let's try to keep naming uniform.

Reply:
done